flox/core.py

```diff
@@ -853,7 +862,8 @@ def _finalize_results(
     """
     squeezed = _squeeze_results(results, axis)
 
-    if agg.min_count > 0:
+    min_count = agg.min_count if agg.min_count is not None else 0
```

Comment:
I'd like to avoid `agg.min_count` being non-integer. There must be a bug somewhere?

Reply:
Probably have to rethink `Aggregation.min_count` then:
https://github.com/xarray-contrib/flox/blob/6cf315a57e006dc01c801039833351240c4431fe/flox/aggregations.py#L210

Reply:
Can we do `self.min_count : int = 0`, hopefully that's easy to propagate

Reply:
This is as far as I can go with min_count=0. It seems xarray_reduce and group_reduce needs to still be `int | None`.

This is the tricky part:
https://github.com/xarray-contrib/flox/blob/6cf315a57e006dc01c801039833351240c4431fe/flox/core.py#L1930-L1938

Reply:
Ah too bad. Thanks for trying!